---
title: "Assignment"
author: "Shobhita"
date: "04/09/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}

install.packages('Seurat')
library(devtools)
library(Seurat)
library(dplyr)
library(Matrix)

# Load the single-cell dataset
csf.data=read.table('GSE127969_counts_TPM_ALL.csv/GSE127969_counts_TPM_ALL.csv',header=T,row.names=1,sep='\t')
csf.data=log(csf.data+1)
celltypes=unlist(lapply(colnames(csf.data), ExtractField, 1))
table(celltypes)


raw_counts<-read.table('GSE127969_counts_TPM_ALL.csv/GSE127969_counts_TPM_ALL.csv',header = TRUE,row.names = 1)

head(raw_counts)

mydata <- CreateSeuratObject(counts = raw_counts, min.cells = 3, min.genes = 200, project = "mydata_scRNAseq")

mydata

mydata[["percent.mt"]] <- PercentageFeatureSet(mydata, pattern = "^MT-")


# Visualize QC metrics as a violin plot
VlnPlot(mydata, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)


mydata <- subset(mydata, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 5)



mydata <- NormalizeData(mydata, normalization.method = "LogNormalize", scale.factor = 10000)


mydata <- NormalizeData(mydata)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
